main.go

```diff
@@ -35,7 +38,7 @@ func main() {
 	must(file.Close())
 
 	if len(cfg.Binaries) == 0 {
-		//lint:ignore ST1005 file is called like this
+		//nolint:stylecheck
```

Comment:
I don't think we want to ignore everything about this line even if the change that something else will be found is very slim. Did more things trigger than stylechecks ST1005?

Reply:
Only `ST1005` was triggered and we can only ignore at the linter level not a specific check within a linter.

Reply:
> Only ST1005 was triggered and we can only ignore at the linter level not a specific check within a linter.

Really? Is that a limitation of golangci-lint? Real bummer and we probably would need to update some comments across the repos.

Reply:
Yeah, unfortunately, it is a bummer. Maybe, the `nolint` directive will be changed in the future to be more specific.

As for your concern

> I don't think we want to ignore everything about this line even if the change that something else will be found is very slim.

You're right, we don't want to ignore everything. However, one thing worth mentioning is that when you run `staticcheck` directly you are running all checks including the ones under [`simple`](https://staticcheck.io/docs/checks/#S) and [`stylecheck`](https://staticcheck.io/docs/checks/#ST).

However, `golangci-lint` distinguishes between these different checks because it doesn't call `staticcheck` directly but instead uses its API. This is also why I have enabled `gosimple` and `stylecheck` linters.

So when we do `nolint:stylecheck` we are not excluding all checks but only those concerned with stylistic issues (`ST` checks) and the checks concerning correctness of code (`SA`) and code simplicity (`S`) will still be applied.

It is ok to use `nolint:stylecheck` for most cases, albeit not ideal. Where possible, we can always make a [slight difference](https://github.com/sapcc/go-makefile-maker/pull/6/commits/2e7d50cfec9cd1efbb563177360a8ee24f35aa7a) and avoid the `nolint` directive altogether.

Reply:
Sounds like an acceptable tradeoff to me.